src/options.c

```diff
@@ -1531,6 +1531,9 @@ read_repo_config_option(char *name, size_t namelen, char *value, size_t valuelen
 	else if (!strcmp(name, "core.abbrev"))
 		parse_int(&opt_id_width, value, 0, SIZEOF_REV - 1);
```

Comment:
Not really important: we have two places where we call
`parse_int()` with a max of `SIZEOF_REV - 1`.
The effective max should maybe be lower for sha1 repos.
But I'm not sure if that  would cause unnecessary friction, so better keep it.

Reply:
My initial code used hash_len but we haven't read extension.objectFormat yet here, so we can't use it. I'll improve this part so that the same configuration can be used in global files for both sha1 and sha256 repositories.